File: projects/CampusAssessment/Models/Talk.cs
using System;
using System.Collections.Generic;

namespace CampusAssessment.Models
{
    public class Talk
    {
        public String id { get; set; }
        public String title { get; set; }
        public String caption { get; set; }
        public String text { get; set; }
        public Speaker speaker { get; set; }
        public List<Review> reviews { get; set; } = new List<Review>();
    }
}

File: projects/CampusAssessment/wwwroot/app/talks/talk.ts
export class Speaker {
    constructor(
        public name: string,
        public email: string,
        public company: string
    ) {}

    toString(): string { // TODO override not working
        return `${this.name} <${this.email}> from ${this.company}`
    }
}

export class Talk {
    constructor (
        public id: string,
        public title: string,
        public caption: string,
        public text: string,
        public speakers: Speaker,
        public reviews: string[]
    ){}
}
